sample-app/src/BaseRouter.tsx

```diff
@@ -0,0 +1,35 @@
+import { ComponentType } from 'react';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+
+interface RouteData {
+  path: string
+  page: JSX.Element
+  exact?: boolean
+}
+
+interface PageProps {
+  Layout?: ComponentType<{ page: JSX.Element }>
+  routes: RouteData[]
+}
```

Comment:
Would it make sense to have a Layout interface defined in typescript? Then the Layout.tsc could implement that and we'd maybe want to rename that Layout since it is a concrete implemenation of the interface?

Reply:
added!